core/src/main/java/io/paradoxical/cassieq/dataAccess/MessageDeletorJobProcessorImpl.java

```diff
@@ -36,31 +37,29 @@
 
 public class MessageDeletorJobProcessorImpl implements MessageDeleterJobProcessor {
     private final Session session;
-    private final QueueRepository queueRepository;
     private final MetricRegistry metricRegistry;
     private final DeletionJob job;
     private final PointerRepository pointerRepository;
     private final MonotonicRepository monotonicRepository;
 
     private static final Logger logger = getLogger(MessageDeletorJobProcessorImpl.class);
+    private final QueueRepository queueRepository;
 
     @Inject
     public MessageDeletorJobProcessorImpl(
             Session session,
-            PointerRepoFactory pointerRepoFactory,
-            MonotonicRepoFactory monotonicRepoFactory,
-            QueueRepository queueRepository,
+            DataContextFactory dataContextFactory,
             MetricRegistry metricRegistry,
             @Assisted DeletionJob job) {
         this.session = session;
-        this.queueRepository = queueRepository;
         this.metricRegistry = metricRegistry;
         this.job = job;
 
-        QueueId queueId = QueueId.valueOf(job.getQueueName(), job.getVersion());
+        QueueId queueId = job.getQueueIdentifier();
 
-        pointerRepository = pointerRepoFactory.forQueue(queueId);
-        monotonicRepository = monotonicRepoFactory.forQueue(queueId);
+        queueRepository = dataContextFactory.forAccount(job.getAccountName());
```

Comment:
this works out really nice now. it was always shitty to have the queue repo be otuside of the context factory. i like that its unified now